### ISSUE TITLE
Switching from Staging to Production acme not working

### ISSUE THREAD

**Issue by valentin-ed** [traefik/traefik#8311]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.containo.us) and didn't find any.

### What did you do?

I use traefik on Docker and i setted up two "letsencrypt" resolver.
The first one for production and the second one for "Staging".
One is writting file to "acme-prod.json" and the other one "acme-staging.json".

I use in my docker label "le-prod" and "le-staging". 

When i first creat my app i set "le-staging" as certificate resolver. When all is good i switch to "le-prod" and recreate my container. 

I suppose the certificate, which is not in the file "acme-prod.json", would be regenerate at this moment. 

### What did you see instead?

Unfortunately my certificate is not renewed and traefik use the one in "staging" file.

### What version of Traefik are you using?

Version:      2.4.12
Codename:     livarot
Go version:   go1.16.6
Built:        2021-07-26T16:16:11Z
OS/Arch:      linux/amd64

### What is your environment & configuration?

```yaml
certificatesResolvers:
  le-prod:
    acme:
      email: "certificate@mydomain.com"
      storage: "/letsencrypt/acme-prod.json"
      tlsChallenge: {}
  le-staging:
    acme:
      caServer: https://acme-staging-v02.api.letsencrypt.org/directory
      email: "certificate@mydomain.com"
      storage: "/letsencrypt/acme-staging.json"
      tlsChallenge: {}
```

**Comment by rtribotte** [comment]
Hello @valentin-ed,

Thanks for your interest in Traefik!

Can you share the debug logs of Traefik, and also the static configuration and the configuration on the containers (both for staging and prod)?

**Comment by EnigmaCurry** [comment]
Hi, I've confirmed this on Traefik 2.4.12 running in docker. 

 * [My static configuration is here](https://github.com/EnigmaCurry/d.rymcg.tech/blob/master/traefik/docker-compose.yaml#L20): 
      * Copy `.env-dist` to `.env` edit it and then run `docker-compose up -d`
 * Two certresolvers are created `staging` and `production`.
 * [I created this whoami container](https://github.com/EnigmaCurry/d.rymcg.tech/tree/master/whoami)
      * Copy `.env-dist` to `.env` edit it and then run `docker-compose up -d`
 * [Here is the debug log while whoami was on the staging certresolver](https://gist.githubusercontent.com/EnigmaCurry/e86d1efbcfb1324319d654ce673e187a/raw/5cc74999dbd8bd94a39a065c429edb53401ba49c/01-traefik-log-staging.txt) 
 * Edit the whoami `.env` file and change `staging` to `production` and redploy with `docker-compose up -d`.
 * [Here is the debug log after whoami is changed to the production certresolver](https://gist.githubusercontent.com/EnigmaCurry/e86d1efbcfb1324319d654ce673e187a/raw/5cc74999dbd8bd94a39a065c429edb53401ba49c/02-traefik-log-production.txt)

The traefik log says this:

```
No ACME certificate generation required for domains [\"whoami.d5.rymcg.tech\"]." providerName=production.acme routerName=whoami@docker rule="Host(`whoami.d5.rymcg.tech`)
```

However the acme-production.json file is empty:

```
$ docker exec -it traefik ls -lha /data
total 24K    
drwxr-xr-x    2 root     root        4.0K Jul 28 16:00 .
drwxr-xr-x    1 root     root        4.0K Jul 28 16:00 ..
-rw-------    1 root     root           0 Jul 28 16:00 acme-production.json
-rw-------    1 root     root       15.8K Jul 28 16:00 acme-staging.json
```

**Comment by rtribotte** [comment]
Hello @valentin-ed @EnigmaCurry,

Thanks @EnigmaCurry for the detailed reproduction case.

I can confirm this behavior and it is due to the fact that, in the end, all certificates are stored in the same default certificate store.
Because of that, when the acme provider is trying to acquire a new certificate and to search for an existing certificate in the only existing store (the default one), it will find the staging certificate (which is valid).

The new certificate resolver configuration would only be taken into account and used if the certificate is expired.

The current workaround is to restart Traefik and comment out the certificate resolver you don't want to use in the static configuration.

The issue you are facing would be fixed by proposal #4756.

So I'll close this issue as a duplicate, but feel free to interact in the original issue.
